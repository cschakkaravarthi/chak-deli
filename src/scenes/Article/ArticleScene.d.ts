import React, { FC } from 'react';
import Article from '../../shared/models/Article.model';
import { match } from 'react-router';
declare type Props = {
    match?: match;
    article?: Article;
    fetchArticleById?: <T>(a: number) => Promise<T>;
    articleId?: number;
};
export declare const ArticleScene: FC<Props>;
declare const _default: import("react-redux").ConnectedComponent<React.FC<Props>, Pick<Props, never> & Props>;
export default _default;
//# sourceMappingURL=ArticleScene.d.ts.map