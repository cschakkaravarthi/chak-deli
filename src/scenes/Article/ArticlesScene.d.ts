import React, { FC } from 'react';
import Article from '../../shared/models/Article.model';
import Facet from '../../shared/models/Facet.model';
declare type Props = {
    facets?: Facet;
    isLastPage: boolean;
    setFilterFacet?: any;
    selectedFacet?: string;
    articleList?: Article[];
    clearArticlesList?: any;
    fetchArticles?: (a?: number, b?: string, c?: string) => void;
    likeArticleToggle?: (contentId: number, likedByUser: boolean) => void;
};
export declare const ArticlesScene: FC<Props>;
declare const _default: import("react-redux").ConnectedComponent<React.FC<Props>, Pick<Props, "facets" | "isLastPage" | "selectedFacet" | "articleList">>;
export default _default;
//# sourceMappingURL=ArticlesScene.d.ts.map