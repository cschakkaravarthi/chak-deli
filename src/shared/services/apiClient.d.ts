declare const client: import("axios").AxiosInstance;
export default client;
//# sourceMappingURL=apiClient.d.ts.map