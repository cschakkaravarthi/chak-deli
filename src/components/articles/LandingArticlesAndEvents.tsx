import React, { FC, useState, useEffect } from 'react';
import get from 'lodash.get';
import Article from '../../shared/models/Article.model';
import { AppState } from '../../shared/types/genericTypes';
import { connect } from 'react-redux';
import { REQUEST_LIMIT_3, SORT_LATEST } from '../../constants/constants';
import { fetchHomeArticles, likeArticleToggle } from '../../shared/actions/articleActions';
import { formatDate } from '../../shared/services/date';
import { Link } from 'react-router-dom';
import createDOMPurify from 'dompurify';
import { Card } from 'umgc_ui_library';
import Col from 'react-bootstrap/Col';
import Row from 'react-bootstrap/Row';
import EventsContainer from '../events/Events';
import EventTeaser from '../events/EventTeaser';
import EventModel from '../../shared/models/Event.model';
import { fetchHomeEvents } from '../../shared/actions/eventActions';

type Props = {
  ownerId?: string;
  articleList?: Article[];
  eventsList?: EventModel[];
  type?: string;
  fetchHomeEvents?: <T>(ownerId: string) => Promise<T>;
  likeArticleToggle?: <T>(contentId: number, likedByUser: boolean) => Promise<T>;
  fetchHomeArticles?: <T>(limit: string, sort: string, ownerId: string) => Promise<T>;
};

const mapState = (state: AppState, props: Props): Props => ({
  ...props,
  eventsList: state.eventReducers.eventsList,
  articleList: state.articleReducers.articleList
});

const actionCreators = { fetchHomeArticles, fetchHomeEvents, likeArticleToggle };

export const LandingArticlesAndEvents: FC<Props> = props => {
  const { articleList, fetchHomeArticles, eventsList, fetchHomeEvents, likeArticleToggle, ownerId = '', type = '' } = props;

  const [articlesFetching, setArticlesFetching] = useState<boolean>(false);
  const [eventsFetching, setEventsFetching] = useState<boolean>(false);

  let otherArticles: Article[] = [];
  let featuredArticle = {} as Article;

  if (articleList) {
    featuredArticle = articleList[0];
    otherArticles = articleList.slice(1, Number(REQUEST_LIMIT_3));
  }

  const articlesAndEventsCall = (): void => {
    setEventsFetching(true);
    setArticlesFetching(true);
    fetchHomeEvents!(ownerId).then(() => setEventsFetching(false));
    fetchHomeArticles!(REQUEST_LIMIT_3, SORT_LATEST, ownerId).then(() => setArticlesFetching(false));
  };

  useEffect(() => {
    articlesAndEventsCall();
  }, []);

  const featuredArticleProps = {
    isFetching: articlesFetching,
    type: 'article',
    variant: 'featured',
    title: get(featuredArticle, 'title'),
    summary: (
      <span
        dangerouslySetInnerHTML={{
          __html: createDOMPurify.sanitize(get(featuredArticle, 'summary', ''))
        }}
      />
    ),
    imageUrl: get(featuredArticle, 'image_uri.umgc_thumbnail'),
    date: formatDate(get(featuredArticle, 'created')),
    categories: get(featuredArticle, 'articleCategory'),
    MainLink: (p: React.PropsWithChildren<{}>) => (
      <Link to={'article/' + get(featuredArticle, 'drupal_id')}>{p.children}</Link>
    ),
    handleThumbClick: () =>
      likeArticleToggle && likeArticleToggle(featuredArticle.drupal_id, !featuredArticle.likedByUser),
    likedStatus: get(featuredArticle, 'likedByUser'),
    likes: get(featuredArticle, 'likesCount')
  };

  return (
    <>
      <Row>
        <Col xl="12" className={type === 'label' ? 'd-none d-xl-block' : 'my-3 d-none d-xl-block'}>
          {!articlesFetching && !featuredArticle ? null : (
            <Card pictureSkeletonHeight={300} {...featuredArticleProps} />
          )}
        </Col>
      </Row>
      <Row>
        <Col xl="6" className="my-3">
          <Row className="h-100">
            <Col md="4" xl="12" className="d-xl-none">
              <Card {...featuredArticleProps} />
            </Col>
            {!articlesFetching && otherArticles.length
              ? otherArticles.map(article => (
                <Col md="4" xl="6" key={article.drupal_id}>
                  <Card
                    type="article"
                    title={article.title}
                    imageUrl={get(article, 'image_uri.umgc_thumbnail', '')}
                    date={formatDate(article.created)}
                    categories={article.articleCategory}
                    MainLink={(p: React.PropsWithChildren<{}>) => (
                      <Link to={'article/' + article.drupal_id}>{p.children}</Link>
                    )}
                    handleThumbClick={() =>
                      likeArticleToggle && likeArticleToggle(article.drupal_id, !article.likedByUser)
                    }
                    likedStatus={article.likedByUser}
                    likes={article.likesCount}
                  />
                </Col>
              ))
              : !articlesFetching && !otherArticles.length
                ? null
                : Array(2)
                  .fill(null)
                  .map((x: '', i: number) => (
                    <Col md="4" xl="6" key={`${i + x}`}>
                      <Card title="" type="article" isFetching={true} />
                    </Col>
                  ))}
          </Row>
        </Col>
        <Col xl="6" className="my-3">
          <EventsContainer events={eventsList} isFetching={eventsFetching}>
            <EventTeaser events={eventsList} isFetching={eventsFetching} />
          </EventsContainer>
        </Col>
      </Row>
    </>
  );
};

export default connect(mapState, actionCreators)(LandingArticlesAndEvents);
