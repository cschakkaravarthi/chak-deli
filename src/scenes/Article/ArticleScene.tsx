import React, { FC, useEffect, useState } from 'react';
import { connect } from 'react-redux';
import createDOMPurify from 'dompurify';
import { fetchArticleById } from '../../shared/actions/articleActions';
import { AppState } from '../../shared/types/genericTypes';
import get from 'lodash.get';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import { formatDate } from '../../shared/services/date';
import { Card } from 'umgc_ui_library/lib';
import { match } from 'react-router';
import { TaxonomyTermModel, VariousContentModel } from '../../shared/types/contentTypes';
import Error404 from '../Errors/Error404';
import { findContentInState } from '../../utils/content';

type Props = {
  match?: match;
  content?: VariousContentModel;
  fetchArticleById?: <T>(a: number) => Promise<T>;
  articleId?: number;
};

const mapState = (state: AppState, props: Props): Props => {
  const articleId = parseInt(get(props, 'match.params.id'), 10);
  // If we clicked on this article from a previous list, then the article already exists in redux. Load it!
  // @TODO: Monitor performance of this solution. Maybe once we find them we should add them to a new set that is
  // @TODO: indexed by drupal_id for quicker lookup?
  const preLoadedArticles = findContentInState(state.contentReducers, [articleId]);
  return {
    ...props,
    articleId,
    content: preLoadedArticles[0]
  };
};

const actionCreators = {
  fetchArticleById
};

export const ArticleScene: FC<Props> = props => {
  const { content, fetchArticleById, articleId } = props;

  const [isFetching, setIsFetching] = useState<boolean>(true);

  if (!articleId) {
    if (!isFetching) {
      return <Error404 />;
    } else {
      return null;
    }
  }

  useEffect(() => {
    if (!content) {
      if (fetchArticleById) {
        fetchArticleById(articleId)
          .then(() => {
            setIsFetching(false);
          })
          .catch(() => {
            setIsFetching(false);
          });
      }
    }
  }, []);

  if (!content) {
    if (!isFetching) {
      return <Error404 />;
    } else {
      return null;
    }
  }

  const fullBody = get(content, 'body_full', '');

  return (
    <Container className="content-full">
      <Row>
        <Col md={12} className="p-1 mt-5">
          <Card
            type="article"
            variant="full"
            title={content.title}
            date={formatDate(get(content, 'created'))}
            imageUrl={get(content, 'image_uri.umgc_featured', '')}
            categories={get(content, 'articleCategory', [] as TaxonomyTermModel[])}
            content={
              <div
                className="position-relative"
                dangerouslySetInnerHTML={{
                  __html: createDOMPurify.sanitize(fullBody)
                }}
              />
            }
          />
        </Col>
      </Row>
    </Container>
  );
};

export default connect(mapState, actionCreators)(ArticleScene);
