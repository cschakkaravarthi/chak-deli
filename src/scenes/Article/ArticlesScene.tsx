import React, { FC, useEffect, useState } from 'react';
import { connect } from 'react-redux';
import { filterQuery } from '../../utils/customHooks';
import { AppState } from '../../shared/types/genericTypes';
import {
  ContentQueryParamsType,
  contentToArticle,
  FacetsModel,
  skeletonBaseContentModel,
  TaxonomyTermModel,
  VariousContentModel
} from '../../shared/types/contentTypes';
import { toQueryString } from '../../utils/content';
import {
  fetchContent,
  fetchContentType,
  likeContentToggle,
  likeContentToggleType
} from '../../shared/actions/contentActions';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import { Form } from 'react-bootstrap';
import { FacetProperty } from '../../shared/models/Facet.model';
import get from 'lodash.get';
import { formatDate } from '../../shared/services/date';
import { Link } from 'react-router-dom';
import { handleScrollToBottom } from '../../utils/scroller';
import { Card } from 'umgc_ui_library/lib';
import * as queryString from 'querystring';
import Skeleton from 'react-loading-skeleton';

const ARTICLE_LIMIT = 9;
const ARTICLE_FACET = 'article_category';
const ARTICLE_SORT = 'latest';

type Props = {
  facets?: FacetsModel;
  moreContentAvailable?: boolean;
  paramSet?: ContentQueryParamsType;
  content?: VariousContentModel[];
  fetchContent: fetchContentType;
  likeContentToggle?: likeContentToggleType;
};

const mapState: any = (state: AppState, props: Props): Props => {
  // First we put together a set of parameters that uniquely identify this data set. This should include any type
  // restrictions, categories, and sorts. Any time I make a query with this parameter set, I should be able to trust
  // that all the data is the same. It should NOT include paging params like limit and offset. We will stringify this
  // param set and use it as the redux key for this set of data, as well as the query string for the API call.
  // @see the ContentQueryParamsType definition for what possible keys should be. Values are comma-separated.
  const paramSet: ContentQueryParamsType = {
    type: 'article',
    facet: ARTICLE_FACET,
    sort: ARTICLE_SORT
  };
  // When a facet has been selected, it will exist in the query string for this scene. Here we load it and apply it to
  // the param set. That means this will be a whole new data set at the new stringified key.
  const chosenFacet = queryString.parse(get(props, 'location.search').substr(1))[ARTICLE_FACET];
  if (chosenFacet && typeof chosenFacet === 'string') {
    paramSet[ARTICLE_FACET] = chosenFacet;
  }
  // Here we get whatever is already loaded at the key that matches the data we require (stringified paramSet).
  const content = state.contentReducers.content[toQueryString(paramSet)];
  return {
    ...props,
    paramSet,
    content: content ? content.content : [],
    facets: content ? content.facets : undefined,
    // If the amount of content we have loaded is less than the total amount of content available for this paramSet,
    // then the pager should be active.
    moreContentAvailable: content ? content.content.length < content.totalRecords : true
  };
};

const actionCreators = {
  fetchContent,
  likeContentToggle
};

export const ArticlesScene: FC<Props> = props => {
  const { content, moreContentAvailable, facets, fetchContent, likeContentToggle, paramSet } = props;

  // This will be a tid if a facet has been selected in the query string, otherwise undefined.
  const selectedFacet = paramSet && paramSet[ARTICLE_FACET];
  const categories = get(facets, ARTICLE_FACET);

  const [isFetching, setIsFetching] = useState(true);

  const enableFetch = (): void => setIsFetching(true);
  const disableFetch = (): void => setIsFetching(false);

  // This method runs constantly as you scroll. If the pager is active, and we aren't actually fetching anything atm,
  // then check to see if the current position is "bottom", and if so, green-light the next "fetch".
  function handleScroll (): void {
    if (moreContentAvailable && !isFetching) {
      handleScrollToBottom(enableFetch);
    }
  }

  // Here we add the scroll listener. Every time isFetching changes, the listener is removed and re-added
  // with the new value.
  useEffect(() => {
    window.addEventListener('scroll', handleScroll);
    return () => {
      window.removeEventListener('scroll', handleScroll);
    };
  }, [isFetching]);

  // Here is where we actually fetch the content. When we have a green-light to fetch (i.e. isFetching is true), then
  // pass our param set to fetchContent, along with our pager parameters to get the next page of content for this
  // paramSet. When it is complete, then disable fetching, so we don't start fetching in a loop. It will stay disabled
  // unless someone scrolls to the bottom and the listener re-enables it. At which point this will fire again.
  useEffect(() => {
    if (isFetching && fetchContent) {
      fetchContent(paramSet || {}, ARTICLE_LIMIT, content ? content.length : 0, disableFetch);
    } else {
      disableFetch();
    }
  }, [isFetching, selectedFacet]);

  // When the drop-down changes, we change the selected facet. That means we are now loading content for a new ParamSet!
  // So we will start with 0 items loaded, and load a whole new set. We assume that sorting, and available filters, and
  // number of items, could all be different for this new ParamSet.
  const setFilterQuery = filterQuery();
  function handleDropDownChange (value: string): void {
    // When the dropdown changes, redirect to the faceted data set.
    setFilterQuery.useFilterQuery(ARTICLE_FACET, value, enableFetch);
  }

  // For this component, it seems reasonable that while we are waiting on content to load we want to show a whole page
  // of skeleton cards. So we create 9 items of "empty" data to feed into the cards, and the UI Library should handle
  // skeletons from there.
  const skeletonContent = [];
  for (let i = 0; i < ARTICLE_LIMIT; i++) {
    const skeletonItem: VariousContentModel = {
      ...skeletonBaseContentModel,
      drupal_id: i,
      type: 'article'
    };
    skeletonContent.push(skeletonItem);
  }

  const loadingContent = (!content || !content.length) && isFetching;

  // Our article set will either be loaded content, or a set of skeletons.
  const articles = loadingContent ? skeletonContent : content;

  return (
    <Container className="article-list mt-5">
      <Row className="mb-2 article-list-header">
        <Col md={6}>
          <p className="font-size-xl">
            <b>News &amp; Articles</b>
          </p>
        </Col>
        <Col md={6} className="article-filter-section my-auto">
          {(categories && categories.length) || !isFetching ? (
            <Form className="d-flex justify-content-start justify-content-md-end">
              <Form.Label className="font-size-sm w-25 d-flex justify-content-start justify-content-md-end mr-sm-3 mr-0 mb-0 align-items-center">
                Filter by:
              </Form.Label>
              <Form.Control
                size="sm"
                as="select"
                id="filter"
                name="filter"
                className="w-auto rounded-0"
                onChange={(e: React.SyntheticEvent) => handleDropDownChange((e.target as HTMLInputElement).value)}
                value={selectedFacet}
              >
                <option value="">All</option>
                {categories instanceof Array &&
                  categories.map((category: FacetProperty) => (
                    <option key={category.drupal_id} value={category.drupal_id}>
                      {category.title}
                    </option>
                  ))}
              </Form.Control>
            </Form>
          ) : (
            <Skeleton width={100} />
          )}
        </Col>
      </Row>
      <Row>
        {articles &&
          articles.map((c: VariousContentModel) => {
            // In this case I expect all content to be of type article, so I convert my type, so I can use properties that
            // are specific to articles, like "likesCount". If I got back some content that didn't match the user type, it
            // would be null, so would not get rendered.
            const article = contentToArticle(c);
            return (
              article && (
                <Col md={4} key={article.drupal_id} className="mb-3 mb-md-5">
                  <Card
                    pictureSkeletonHeight={300}
                    isFetching={loadingContent}
                    type="article"
                    title={article.title}
                    imageUrl={get(article, 'image_uri.umgc_thumbnail', '')}
                    date={formatDate(article.created)}
                    categories={get(article, 'articleCategory', [] as TaxonomyTermModel[])}
                    MainLink={(p: React.PropsWithChildren<{}>) => (
                      <Link to={'article/' + article.drupal_id}>{p.children}</Link>
                    )}
                    handleThumbClick={() =>
                      likeContentToggle && likeContentToggle(article.drupal_id, !article.likedByUser)
                    }
                    likedStatus={article.likedByUser}
                    likes={article.likesCount}
                  />
                </Col>
              )
            );
          })}
      </Row>
      <Row>
        <Col>
          {isFetching && (
            <Col className="p-5 d-flex justify-content-center text-secondary">
              <strong>Fetching articles ...</strong>
            </Col>
          )}
        </Col>
      </Row>
    </Container>
  );
};

export default connect(mapState, actionCreators)(ArticlesScene);
