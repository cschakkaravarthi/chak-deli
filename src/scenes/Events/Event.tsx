import React, { FC, useEffect, useRef, useState } from 'react';
import { connect } from 'react-redux';
import get from 'lodash.get';
import createDOMPurify from 'dompurify';
import Col from 'react-bootstrap/Col';
import EventModel from '../../shared/models/Event.model';
import Row from 'react-bootstrap/Row';
import { AppState } from '../../shared/types/genericTypes';
import { Card } from 'umgc_ui_library/lib';
import { Container } from 'react-bootstrap';
import { fetchEventById } from '../../shared/actions/eventActions';
import { usehandleAddToCalendar, normalizeEventData } from '../../utils/customHooks';
import { formatDateTime, formatDateRangeWithTime, formatDateRange } from '../../shared/services/date';
import BSToast from '../../components/toast/Toast';
import { match } from 'react-router';
import { TaxonomyTermModel } from '../../shared/types/contentTypes';
import Error404 from '../Errors/Error404';

type Props = {
  match?: match;
  event?: EventModel;
  eventsList?: EventModel[];
  moreEventsList?: EventModel[];
  fetchEventById?: (a: number | string) => Promise<void>;
};

const mapState = (state: AppState, props: Props): Props => (
  {
    ...props,
    event: state.eventReducers.event,
    eventsList: state.eventReducers.eventsList,
    moreEventsList: state.eventReducers.moreEventsList
  }
);

const actionCreators = { fetchEventById };

export const Event: FC<Props> = props => {
  const { match, event, eventsList, moreEventsList, fetchEventById } = props;

  const [isFetching, setIsFetching] = useState<boolean>(true);

  const urlEventId = parseInt(get(match, 'params.id'), 10);

  // eventsList stores the events on home page
  // while moreEventsList the events in the events scene.
  const eventStore = moreEventsList || eventsList;

  if (!urlEventId) {
    if (!isFetching) {
      return <Error404 />;
    } else {
      return null;
    }
  }

  const shouldFetch = eventStore
    ? eventStore.find(eventItem => eventItem.drupal_id === urlEventId)
    : undefined;

  useEffect(() => {
    if (!shouldFetch) {
      if (fetchEventById) {
        fetchEventById(urlEventId)
          .then(() => {
            setIsFetching(false);
          })
          .catch(() => {
            setIsFetching(false);
          });
      }
    } else {
      setIsFetching(false);
    }
  }, []);

  const eventResolution = shouldFetch || event;

  const toastRef = useRef();

  const handleAddToCalender = (id: string, title: string): void => {
    usehandleAddToCalendar(id, title, toastRef);
  };

  if (!eventResolution) {
    if (!isFetching) {
      return <Error404 />;
    } else {
      return null;
    }
  }

  const getDate = (event: EventModel): string => {
    const limit = 23 * 60 * 60 + 59 * 60; // 23 hours and 59 seconds
    if (event.when_end) {
      const difference = event.when_end - event.when_start;

      if (difference < limit) {
        return formatDateRangeWithTime(event.when_start, event.when_end);
      } else if (difference === limit) {
        return formatDateTime(event.when_start, 'noTZ');
      } else {
        return formatDateRange(event.when_start, event.when_end);
      }
    } else {
      return formatDateTime(event.when_start, 'noTZ');
    }
  };

  return (
    <Container className="content-full">
      <Row>
        <Col md={12} className="p-0 mt-5">
          <BSToast ref={toastRef}/>
          <Card
            type="event"
            title={eventResolution.title}
            variant="full"
            location={eventResolution.where}
            date={getDate(eventResolution)}
            imageUrl={get(eventResolution, 'image_uri.umgc_featured', '')}
            categories={get(
              eventResolution,
              'event_category',
              [] as TaxonomyTermModel[]
            )}
            content={
              <div
                className="position-relative"
                dangerouslySetInnerHTML={{
                  __html: createDOMPurify.sanitize(
                    get(eventResolution, 'body_full', '')
                  )
                }}
              />
            }
            AddToCalenderLink={() => {
              handleAddToCalender(
                eventResolution.drupal_id.toString(),
                eventResolution.title
              );
            }}
            SendInviteEventData = {normalizeEventData(eventResolution)}
          />
        </Col>
      </Row>
    </Container>
  );
};

export default connect(
  mapState,
  actionCreators
)(Event);
