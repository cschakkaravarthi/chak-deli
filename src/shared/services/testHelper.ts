import { ReactElement } from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import getStore from './mockGlobalStore';

export const shallowRender = (toRender: JSX.Element): ReactElement => {
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(toRender);
  return renderer.getRenderOutput();
};

export const getMockStore = getStore;

export const drawerMockNavigation = {
  state: {
    isDrawerOpen: true,
    isTransitioning: false,
    index: 1,
    routes: [],
    key: 'abc',
    params: {}
  },
  dispatch: jest.fn(),
  goBack: jest.fn(),
  dismiss: jest.fn(),
  navigate: jest.fn(),
  openDrawer: jest.fn(),
  closeDrawer: jest.fn(),
  toggleDrawer: jest.fn(),
  getParam: jest.fn(),
  setParams: jest.fn(),
  addListener: jest.fn(),
  push: jest.fn(),
  replace: jest.fn(),
  reset: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
  isFocused: jest.fn(),
  isFirstRouteInParent: jest.fn(),
  dangerouslyGetParent: jest.fn()
};
