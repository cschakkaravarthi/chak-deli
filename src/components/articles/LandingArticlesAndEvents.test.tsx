import React from 'react';
import { shallowRender } from '../../shared/services/testHelper';
import { LandingArticlesAndEvents } from './LandingArticlesAndEvents';
import { dummyArticleList } from '../../shared/models/Article.model';
import { dummyEvents } from '../../shared/models/Event.model';

describe('LandingArticlesAndEvents', () => {
  it('null on articles and events components', () => {
    const tree = shallowRender(
      <LandingArticlesAndEvents eventsList={[]} articleList={[]} />
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders articles and events', () => {
    const tree = shallowRender(
      <LandingArticlesAndEvents eventsList={dummyEvents} articleList={dummyArticleList} />
    );
    expect(tree).toMatchSnapshot();
  });
});
