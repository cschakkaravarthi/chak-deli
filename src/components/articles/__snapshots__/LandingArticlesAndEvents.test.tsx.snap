// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingArticlesAndEvents null on articles and events components 1`] = `
<React.Fragment>
  <ForwardRef
    noGutters={false}
  >
    <ForwardRef
      className="my-3 d-none d-xl-block"
      xl="12"
    />
  </ForwardRef>
  <ForwardRef
    noGutters={false}
  >
    <ForwardRef
      className="my-3"
      xl="6"
    >
      <ForwardRef
        className="h-100"
        noGutters={false}
      >
        <ForwardRef
          className="d-xl-none"
          md="4"
          xl="12"
        >
          <Unknown
            MainLink={[Function]}
            date=""
            handleThumbClick={[Function]}
            isFetching={false}
            summary={
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "",
                  }
                }
              />
            }
            type="article"
            variant="featured"
          />
        </ForwardRef>
      </ForwardRef>
    </ForwardRef>
    <ForwardRef
      className="my-3"
      xl="6"
    >
      <Unknown
        events={Array []}
        isFetching={false}
      >
        <Unknown
          events={Array []}
          isFetching={false}
        />
      </Unknown>
    </ForwardRef>
  </ForwardRef>
</React.Fragment>
`;

exports[`LandingArticlesAndEvents renders articles and events 1`] = `
<React.Fragment>
  <ForwardRef
    noGutters={false}
  >
    <ForwardRef
      className="my-3 d-none d-xl-block"
      xl="12"
    >
      <Unknown
        MainLink={[Function]}
        categories={
          Array [
            Object {
              "drupal_id": 45,
              "icon": null,
              "title": "Industry News",
              "weight": 1,
            },
          ]
        }
        date="June 20, 2019"
        handleThumbClick={[Function]}
        imageUrl="undefined"
        isFetching={false}
        pictureSkeletonHeight={300}
        summary={
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "An article about music.",
              }
            }
          />
        }
        title="Zeppelin"
        type="article"
        variant="featured"
      />
    </ForwardRef>
  </ForwardRef>
  <ForwardRef
    noGutters={false}
  >
    <ForwardRef
      className="my-3"
      xl="6"
    >
      <ForwardRef
        className="h-100"
        noGutters={false}
      >
        <ForwardRef
          className="d-xl-none"
          md="4"
          xl="12"
        >
          <Unknown
            MainLink={[Function]}
            categories={
              Array [
                Object {
                  "drupal_id": 45,
                  "icon": null,
                  "title": "Industry News",
                  "weight": 1,
                },
              ]
            }
            date="June 20, 2019"
            handleThumbClick={[Function]}
            imageUrl="undefined"
            isFetching={false}
            summary={
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "An article about music.",
                  }
                }
              />
            }
            title="Zeppelin"
            type="article"
            variant="featured"
          />
        </ForwardRef>
        <ForwardRef
          md="4"
          xl="6"
        >
          <Unknown
            MainLink={[Function]}
            categories={
              Array [
                Object {
                  "drupal_id": 45,
                  "icon": null,
                  "title": "Industry News",
                  "weight": 1,
                },
              ]
            }
            date="June 20, 2019"
            handleThumbClick={[Function]}
            imageUrl="undefined"
            title="Floyd"
            type="article"
          />
        </ForwardRef>
        <ForwardRef
          md="4"
          xl="6"
        >
          <Unknown
            MainLink={[Function]}
            categories={
              Array [
                Object {
                  "drupal_id": 45,
                  "icon": null,
                  "title": "Industry News",
                  "weight": 1,
                },
              ]
            }
            date="June 20, 2019"
            handleThumbClick={[Function]}
            imageUrl="undefined"
            title="Hendrix"
            type="article"
          />
        </ForwardRef>
      </ForwardRef>
    </ForwardRef>
    <ForwardRef
      className="my-3"
      xl="6"
    >
      <Unknown
        events={
          Array [
            Object {
              "attachments": null,
              "body_full": "<p>Some body text</p>
",
              "categories": Array [],
              "category_ancestors": Array [],
              "created": 1568326940,
              "drupal_id": 4,
              "event_category": Array [
                Object {
                  "drupal_id": 46,
                  "icon": null,
                  "title": "Industry Events",
                  "weight": 1,
                },
              ],
              "image_uri": Object {
                "source": "undefined",
                "umgc_banner": "undefined",
                "umgc_embedded": "undefined",
                "umgc_featured": "undefined",
                "umgc_thumbnail": "undefined",
              },
              "language": "en",
              "owner": Array [
                Object {
                  "drupal_id": 3,
                  "icon": null,
                  "title": "Global Travel",
                  "weight": 0,
                },
              ],
              "summary": "",
              "tags": Array [
                Object {
                  "drupal_id": 6,
                  "icon": null,
                  "title": "Lady Gaga",
                  "weight": 0,
                },
              ],
              "tags_ancestors": Array [
                Object {
                  "drupal_id": 6,
                  "icon": null,
                  "title": "Lady Gaga",
                  "weight": 0,
                },
              ],
              "title": "Test 1",
              "type": "event",
              "when_duration": 7199,
              "when_end": 1568370600,
              "when_start": 1568326940,
              "where": "Over the Rainbow",
            },
            Object {
              "attachments": null,
              "body_full": "<p>Some body text</p>
",
              "categories": Array [],
              "category_ancestors": Array [],
              "created": 1568326940,
              "drupal_id": 24,
              "event_category": Array [
                Object {
                  "drupal_id": 46,
                  "icon": null,
                  "title": "Industry Events",
                  "weight": 1,
                },
              ],
              "image_uri": Object {
                "source": "undefined",
                "umgc_banner": "undefined",
                "umgc_embedded": "undefined",
                "umgc_featured": "undefined",
                "umgc_thumbnail": "undefined",
              },
              "language": "en",
              "owner": Array [
                Object {
                  "drupal_id": 3,
                  "icon": null,
                  "title": "Global Travel",
                  "weight": 0,
                },
              ],
              "summary": "",
              "tags": Array [
                Object {
                  "drupal_id": 6,
                  "icon": null,
                  "title": "Lady Gaga",
                  "weight": 0,
                },
              ],
              "tags_ancestors": Array [
                Object {
                  "drupal_id": 6,
                  "icon": null,
                  "title": "Lady Gaga",
                  "weight": 0,
                },
              ],
              "title": "Test 1",
              "type": "event",
              "when_duration": 7199,
              "when_end": 1568370600,
              "when_start": 1568326940,
              "where": "Over the Rainbow",
            },
          ]
        }
        isFetching={false}
      >
        <Unknown
          events={
            Array [
              Object {
                "attachments": null,
                "body_full": "<p>Some body text</p>
",
                "categories": Array [],
                "category_ancestors": Array [],
                "created": 1568326940,
                "drupal_id": 4,
                "event_category": Array [
                  Object {
                    "drupal_id": 46,
                    "icon": null,
                    "title": "Industry Events",
                    "weight": 1,
                  },
                ],
                "image_uri": Object {
                  "source": "undefined",
                  "umgc_banner": "undefined",
                  "umgc_embedded": "undefined",
                  "umgc_featured": "undefined",
                  "umgc_thumbnail": "undefined",
                },
                "language": "en",
                "owner": Array [
                  Object {
                    "drupal_id": 3,
                    "icon": null,
                    "title": "Global Travel",
                    "weight": 0,
                  },
                ],
                "summary": "",
                "tags": Array [
                  Object {
                    "drupal_id": 6,
                    "icon": null,
                    "title": "Lady Gaga",
                    "weight": 0,
                  },
                ],
                "tags_ancestors": Array [
                  Object {
                    "drupal_id": 6,
                    "icon": null,
                    "title": "Lady Gaga",
                    "weight": 0,
                  },
                ],
                "title": "Test 1",
                "type": "event",
                "when_duration": 7199,
                "when_end": 1568370600,
                "when_start": 1568326940,
                "where": "Over the Rainbow",
              },
              Object {
                "attachments": null,
                "body_full": "<p>Some body text</p>
",
                "categories": Array [],
                "category_ancestors": Array [],
                "created": 1568326940,
                "drupal_id": 24,
                "event_category": Array [
                  Object {
                    "drupal_id": 46,
                    "icon": null,
                    "title": "Industry Events",
                    "weight": 1,
                  },
                ],
                "image_uri": Object {
                  "source": "undefined",
                  "umgc_banner": "undefined",
                  "umgc_embedded": "undefined",
                  "umgc_featured": "undefined",
                  "umgc_thumbnail": "undefined",
                },
                "language": "en",
                "owner": Array [
                  Object {
                    "drupal_id": 3,
                    "icon": null,
                    "title": "Global Travel",
                    "weight": 0,
                  },
                ],
                "summary": "",
                "tags": Array [
                  Object {
                    "drupal_id": 6,
                    "icon": null,
                    "title": "Lady Gaga",
                    "weight": 0,
                  },
                ],
                "tags_ancestors": Array [
                  Object {
                    "drupal_id": 6,
                    "icon": null,
                    "title": "Lady Gaga",
                    "weight": 0,
                  },
                ],
                "title": "Test 1",
                "type": "event",
                "when_duration": 7199,
                "when_end": 1568370600,
                "when_start": 1568326940,
                "where": "Over the Rainbow",
              },
            ]
          }
          isFetching={false}
        />
      </Unknown>
    </ForwardRef>
  </ForwardRef>
</React.Fragment>
`;
